vision/google/cloud/vision/image.py

```diff
@@ -30,22 +30,35 @@ class Image(object):
     :type content: bytes
     :param content: Byte stream of an image.
 
+    :type filename: str
+    :param filename: Filename to image.
+
     :type source_uri: str
     :param source_uri: Google Cloud Storage URI of image.
 
     :type client: :class:`~google.cloud.vision.client.Client`
     :param client: Instance of Vision client.
     """
 
-    def __init__(self, client, content=None, source_uri=None):
+    def __init__(self, client, content=None, filename=None, source_uri=None):
+        sources = set(source for source in (content, filename, source_uri)
+                      if source is not None)
```

Comment:
As above, don't use `set()`, it's "wasteful" (though negligible, since it is creating a hash table from at most 3 values).